include/benchmark/benchmark.h

```diff
@@ -465,7 +472,7 @@ inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp const& value) {
 }
 
 template <class Tp>
-inline BENCHMARK_ALWAYS_INLINE void DoNotOptimize(Tp& value) {
```

Comment:
But this currently takes a simple reference?
https://godbolt.org/z/z3GP44jnG

Reply:
The magic is in using `template <typename T>`
compiler collapses extra '&'.  if T is alread 'int&', 'T&&' - 'int&&&' transforms into 'int&'. This is universal reference.
see https://godbolt.org/z/eEfxfa8E6